GetDKAN/dkan
# Control datastore via REST API

This adds an endpoint and resources for controling a _resource_ node's datastore via the REST API.

## QA Steps

Using whatever means you prefer to test the API (I like Postman; [here is a collection that includes requests for all these QA steps](https://gist.github.com/dafeder/be673ea8e51845bf3a09645e2a718e97)), test the following actions, using the example of the "Violent Crime Statistics By City" resource:

- [ ] Create datastore configuration: POST /api/dataset/datastore/19:

```json
{
	"manager": "simple",
	"configuration": {
		"delimiter": ",",
		"quote": "\"",
		"escape": "\\"
	} 
}
```

- [ ] Retrieve datastore information: GET /api/dataset/datastore/19

- [ ] Update datastore configuration: PUT /api/dataset/datastore/19: 

```json
{
	"configuration": {
		"delimiter": "|",
		"quote": "\"",
		"escape": "\\"
	} 
}
```
(Also try enabling the fast import module and changing the `manager` property to `fast`; it should refuse)

- [ ] Remove datastore: DELETE /api/dataset/datastore/19
- [ ] Import datastore: POST /api/dataset/datastore/19/import
- [ ] Drop datastore: POST /api/dataset/datastore/19/drop
